### PR TITLE
Hide things from getting accidentally logged in chat

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers

### CODE REVIEW

File: shared/constants/chat.js [hunk 1]
```diff
@@ -28,103 +29,102 @@ export type Message = ClientMessage | ServerMessage
...
-  info: null,
-  participants: List(),
-  conversationIDKey: '',
-  muted: false,
-  time: '',
-  snippet: '',
-  unreadCount: 0,
+export const ConversationStateRecord: Class<ConversationState> = Record({
+  messages: List(),
+  seenMessages: Set(),
```
Comment (chromakode) [inline shared/constants/chat.js]: Should this be hidden as well?

File: shared/actions/chat.js [hunk 2]
```diff
@@ -230,23 +231,24 @@ function * _postMessage (action: PostMessage): SagaGenerator<any, any> {
...
+      const incomingMessage: ?IncomingMessageRPCType = activity.incomingMessage
       if (incomingMessage) {
         const messageUnboxed: MessageUnboxed = incomingMessage.message
         const yourName = yield select(usernameSelector)
         const conversationIDKey = conversationIDToKey(incomingMessage.convID)
         const message = _unboxedToMessage(messageUnboxed, 0, yourName, conversationIDKey)
 
         // TODO short-term if we haven't seen this in the conversation list we'll refresh the inbox. Instead do an integration w/ gregor
-        const conversationStateSelector = (state: TypedState) => state.chat.get('conversationStates', Map()).get(conversationIDKey)
+        const conversationStateSelector: (state: TypedState) => string = (state: TypedState) => state.chat.get('conversationStates')
```
Comment (chromakode) [inline shared/actions/chat.js]: Why is the `.get(conversationIDKey)` no longer necessary?

Comment (chromakode) [inline shared/actions/chat.js]: I'm confused how this type checks... shouldn't `state.chat.get('conversationStates')` be a `Map`?

Comment (MarcoPolo) [inline shared/actions/chat.js]: `.get` doesn't add any type safety

**Pull request closed without merging**